pkg/cmap/cmap.go

```
@@ -18,50 +18,96 @@
+// shardForKey returns the shard for the given key.
+func (m *Map[K, V]) shardForKey(key K) *shard[K, V] {
+	var idx uint32
+	switch k := any(key).(type) {
+	case string:
+		hash := fnv.New32a()
+		if _, err := hash.Write([]byte(k)); err != nil {
+			panic(fmt.Sprintf("shard for key: %s", err))
+		}
+		idx = hash.Sum32()
+	case int:
+		idx = uint32(k)
```

golangci-lint

[high] G115: integer overflow conversion int -> uint32

(gosec)